### PR TITLE
remove repo filtering on the github credentials

### LINKED ISSUE #6537: github-org processor doesnt work with repo limited github app installation
When using the github app integration setup and the app installation has been limited to a subset of repositories then the github-org processor fails with the following error:

```
2021-07-19T15:42:05.084Z catalog warn Processor GithubOrgReaderProcessor threw an error while reading github-org:https://github.com/AcmeInc; caused by Error: The Backstage GitHub application used in the AcmeInc organization must be installed for the entire organization to be able to issue credentials without a specified repository. type=plugin
```

As you can see this github-org processor is failing for a reason that seems to be related to repositories, however github teams dont operate around the concept of repositories.

The docs for gihub apps is here https://backstage.io/docs/plugins/github-apps.

## Current Behavior

The github-org teams are not loaded into the catalog.

```
2021-07-19T15:42:05.084Z catalog warn Processor GithubOrgReaderProcessor threw an error while reading github-org:https://github.com/AcmeInc; caused by Error: The Backstage GitHub application used in the AcmeInc organization must be installed for the entire organization to be able to issue credentials without a specified repository. type=plugin
```

## Possible Solution

I'd like to suggest that the repo filter is no longer sent to github when requesting credentials. This means that there will be a single token cached per owner installation at a time.

## Steps to Reproduce

1. Create a github app
2. Install it to organization and limit the set of repos that are available to that installation
3. Configure the github-org processor for that organization.

## Context

This results in it not being possible to use the github-org processor while limiting the repos visible to backstage.

## Your Environment

- NodeJS Version (v12): node:14-buster-slim docker container image
- Operating System and Version (e.g. Ubuntu 14.04): Alpine Linux
- Browser Information: N/A

### PR DESCRIPTION
**Pull request by punkle**
## Hey, I just made a Pull Request!

The github-org processor does not use the concept of repositories.
As such filtering based on selected repositories does not make sense
for that processer. The GithubCredentials was imposing what appears
like an unnessesary restriction on the tokens using the repo name.

By removing this restriction, it enalbes the github-org processor to
work when a github app installation has a repository restriction.

closes #6537 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey @punkle  - so I've added the `after-vacation` label for this as I'm gonna need some extra eyes on this when the other folks are back from vacation in a few weeks. I'm not sure if this is what we want to do here, thought there was some specific reasons why we did this.

**Comment by punkle** [comment]
@benjdlambert thanks 😊

**Comment by punkle** [comment]
@benjdlambert do you know how to disable that codeql rule for that failure? The code make sense in the context it is in. It is a test, so I think disabling the check is the right thing to do in this case.

**Comment by dtuite** [comment]
@benjaminr @punkle I dismissed the CodeQL error on Brian's behalf since it is not a real secret and is only used in tests.

**Comment by punkle** [comment]
I have found a possible reason why it was done this way. I think there was an attempt to show a reasonable error when the repo was not accessable to the github app installation. I have added an additional api call to list the repositories available to installation and then showing a nicer error in this same case.

I continue to look forward to a review when the team return from vacation.

File: packages/integration/src/github/GithubCredentialsProvider.ts [hunk 1]
```diff
@@ -233,7 +233,10 @@ export class GithubCredentialsProvider {
     const parsed = parseGitUrl(opts.url);
 
     const owner = parsed.owner || parsed.name;
-    const repo = parsed.owner ? parsed.name : undefined;
+    let repo = parsed.owner ? parsed.name : undefined;
+    // the github-discovery plugin passes an • as a repo name. This simply means it wants access to all repos
+    // that are available to the installation.
+    repo = repo === '*' ? undefined : repo;
```
Comment (Rugvip) [inline packages/integration/src/github/GithubCredentialsProvider.ts:239]: Can we skip this? 😅 I'm assuming it's referring to the GitHub discovery processor in the catalog? Thinking the URL would just be `https://github.com/backstage` for when you want to ingest all repos in an org. That or make it clear that this is a feature and document it above.

File: packages/integration/src/github/GithubCredentialsProvider.ts [hunk 2]
```diff
@@ -85,34 +84,36 @@ class GithubAppManager {
...
-    const {
-      installationId,
-      suspended,
-      repositorySelection,
-    } = await this.getInstallationData(owner);
+    const { installationId, suspended } = await this.getInstallationData(owner);
     if (suspended) {
       throw new Error(
-        `The GitHub application for ${[owner, repo]
+        `The GitHub application for ${[owner]
```
Comment (Rugvip) [inline packages/integration/src/github/GithubCredentialsProvider.ts:90]: No need for the array handling now x)

**Review by Rugvip — COMMENTED** [review]
Thanks! I think this makes sense, the reason for the existing implementation was to break early when trying to access org-only APIs, but given that that breaks discovery this should be better

Just a few nits

File: packages/integration/src/github/GithubCredentialsProvider.ts (hunk 2 above)
Comment (punkle) [reply packages/integration/src/github/GithubCredentialsProvider.ts]: done

File: packages/integration/src/github/GithubCredentialsProvider.ts (hunk 1 above)
Comment (punkle) [reply packages/integration/src/github/GithubCredentialsProvider.ts]: I have moved the handling of this special case back into the GitHub discovery processor. Let me know how it looks. I'm not sure how confident I am about how I have chosen to build the GitHub Org URL.

File: plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts [hunk 3]
```diff
@@ -63,9 +64,15 @@ export class GithubDiscoveryProcessor implements CatalogProcessor {
         `There is no GitHub integration that matches ${location.target}. Please add a configuration entry for it under integrations.github`,
       );
     }
+
+    // Building the org url here so that the github creds provider doesn't need to know
+    // about how to handle the wild card which is special for this processor.
+    const { source, organization } = parseGitUrl(location.target);
```
Comment (Rugvip) [inline plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts:70]: `organization` seems to be a CloudForge thing according to docs. Afaik best way to get owner is `owner ?? name`, although in this case you got the `parseUrl` happening just below so perhaps move that up instead?

**Review by jhaals — APPROVED** [review]
I had to dig deep in my memory to remember the reasoning behind the access error but I believe it was to exit early with a nice error message. I think this looks good 👍

**Review by Rugvip — APPROVED** [review]
Yep, :shipit:

**Comment by freben** [comment]
@punkle You may want to backtrack a bit - I see you got bitten by the `prettier` bump. If you merge master into your branch, you will have to run `yarn install` first, and then `yarn prettier --write` to apply all of the proper formatting. In its current state, your branch tries to go back to the old format that was applicable before the `prettier` bump.

**Comment by punkle** [comment]
Thanks @freben, hopefully this latest commit is better.

**Pull request merged**